sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java

```diff
@@ -28,47 +33,131 @@ public NetworkCallError() {
+     * @return The instance of NetworkCallError class.
+     */
+    public NetworkCallError className(Class<?> className) {
+        this.className = className;
+        return this;
+    }
 
-            case "java.lang.IndexOutOfBoundsException":
-                return new IndexOutOfBoundsException(throwable.getMessage());
+    /**
+     * Gets the name of the class of the throwable. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @return The class name which is Class type.
+     */
+    public Class<?> className() {
+        return this.className;
+    }
 
-            case "java.net.UnknownHostException":
-                return new UnknownHostException(throwable.getMessage());
+    /**
+     * Sets the throwable argument types which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @param argTypes The array of argument types.
+     * @return The instance of NetworkCallError class.
+     */
+    public NetworkCallError argTypes(Class<?>[] argTypes) {
+        this.argTypes = ImplUtils.clone(argTypes);
+        return this;
+    }
 
-            default:
-                return throwable;
-        }
+    /**
+     * Gets the throwable argument types which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @return The array of argument types.
+     */
+    public Class<?>[] argTypes() {
+        return ImplUtils.clone(this.argTypes);
     }
 
     /**
-     * Sets the throwable that was thrown during a network call.
+     * Sets the throwable argument default values which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
      *
-     * @param throwable Throwable that was thrown.
+     * @param argValues The array of argument values.
+     * @return The instance of NetworkCallError class.
      */
-    public void throwable(Throwable throwable) {
-        this.throwable = throwable;
+    public NetworkCallError argValues(Object[] argValues) {
+        this.argValues = ImplUtils.clone(argValues);
+        return this;
     }
 
     /**
-     * Sets the name of the class of the throwable. This is used during deserialization the construct the throwable
+     * Gets the throwable argument default values which needed for new instance constructor. This is used during deserialization the construct the throwable
      * as the actual class that was thrown.
      *
-     * @param className Class name of the throwable.
+     * @return The array of argument values.
      */
-    public void className(String className) {
-        this.className = className;
+    public Object[] argValues() {
+        return ImplUtils.clone(this.argValues);
```

Comment:
I don't think this is going to work as you expect, this will clone the array with the same references (aka a shallow copy).

Reply:
ImplUtil will do a deep copy.
The class<?> is able to copy to new place. What we need is the class path.
The object[] , primitive is able to copy. if it is non-primitive the default value we use is null.
Copy will not give us wrong result.
Since not every exception expose the getter to the field, I am not able to get real value but the dummy default one to make it work in record and playback.